api/python/t4/snapshots.py

```diff
@@ -575,8 +630,23 @@ def materialize(self, path, name=None):
             fail to put bytes
             fail to put package to registry
         """
-        raise NotImplementedError
-        if name is None:
-            raise NotImplementedError
-        self.get_files(path)
-        self.dump(get_package_registry(path) + name)
+        pkg = self._clone()
+        for logical_key, entry in self._data.items():
+            # Copy the datafiles in the package.
+            if path.startswith('s3://'):
+                new_physical_key = path + "/" + logical_key
+                new_type = PhysicalKeyType.S3.name
+            else:
+                new_physical_key = os.path.join(path, logical_key)
```

Comment:
These are supposed to be URIs, not paths... Need to parse `path` and re-build it - or just pass the parsed URI into this method. Or, if we know `path` is a URI, then something like `path + '/' + quote(logical_key)` would be correct.

Reply:
Swapped to use resolve as uri. Leaving as a string to path instead of uri for now because this is supposed to help with downloading all the contents of a package, so nice to be able to use a relative path. This will end up getting refactored though once we have a stable user facing way to call materialize.